sklearn/cluster/spectral.py

```diff
@@ -325,13 +325,18 @@ class SpectralClustering(BaseEstimator, ClusterMixin):
 
     affinity : string, array-like or callable, default 'rbf'
         If a string, this may be one of 'nearest_neighbors', 'precomputed',
-        'rbf' or one of the kernels supported by
-        `sklearn.metrics.pairwise_kernels`.
+        'precomputed_nearest_neighbors', rbf', or one of the kernels supported
+        by `sklearn.metrics.pairwise_kernels`.
 
         Only kernels that produce similarity scores (non-negative values that
         increase with similarity) should be used. This property is not checked
         by the clustering algorithm.
 
+        Note that 'precomputed' uses directly ``X`` as the affinity matrix,
+        whereas 'precomputed_nearest_neighbors' filters the affinity matrix
```

Comment:
I find this unclear. maybe: "'precomputed_nearest_neighbors' accepts a sparse graph of precomputed nearest neighbors, and constructs the kernel by selecting the n_neighbors nearest neighbors"...

Reply:
If SpectralClustering supports the `_pairwise` attribute, the 'precomputed_nearest_neighbors' case should be handled there.